qiime2/sdk/usage.py

```diff
@@ -0,0 +1,349 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016-2020, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import abc
+import os
+import re
+import types
+import tempfile
+import uuid
+import zipfile
+
+from qiime2 import sdk
+
+# TODO: docstrings
+
+
+class UsageAction:
+    def __init__(self, plugin_id: str, action_name: str):
+        if plugin_id == '':
+            raise ValueError('Must specify a value for plugin_id.')
+
+        if action_name == '':
+            raise ValueError('Must specify a value for action_name.')
```

Comment:
is this an `action_id`?

Reply:
Good catch - this was part of the "stuff that changed between usage and #511" that I mentioned IRL this morning.